tests/phpunit/tests/test-wpml-gutenberg-integration.php

```diff
@@ -529,6 +529,144 @@ public function it_updates_inner_blocks_with_innerContent() {
 
 	}
 
+	/**
+	 * @test
+	 * @group wpmlcore-6627
+	 */
+	public function it_updates_root_content_and_inner_blocks() {
```

Comment:
I wrote tests in this integration class otherwise we would miss the recursive part of the inner blocks replacement. I tried to detailed test steps as much as I can, but we are testing many things at this point :)

Reply:
Huge test: I'll have to trust you here :)

Reply:
@strategio How about extracting `update_string_in_innerContent` into a separate class that you can inject into this class. Then you only need to test that it's called with the correct arguments. 
Then the function in the new class can be tested separately which should be much simpler.

Reply:
@brucepearson, even if I extract `update_string_in_innerContent`, we would still have to test it from the integration class. The method I am testing here is `WPML_Gutenberg_Integration::string_translated` which is calling `WPML_Gutenberg_Integration::update_block_translations`, a recursive function.

The idea is not only to cover the code (which I could do by a test in `TestHTML`), but to test a real block with inner blocks. And that's why we need to test the recursion part.

Reply:
ok